### DISCUSSION TITLE
using OpenAI SDK to request the LocalAI api

### DISCUSSION THREAD

**Discussion by gadingnst** [mudler/LocalAI#907] [General]
can I use [OpenAI SDK](https://www.npmjs.com/package/openai) to request LocalAI api?
if it is possible, how?

I've already done this, but it doesn't work. the error message is `ECONN Refused`.
any suggestion/advice? 😅

```js
import OpenAIApi from 'openai';

const OpenAI = new OpenAIApi({
  apiKey: 'my key',
  baseURL: 'http://localhost:8080'
});

export default OpenAI;
```

**Comment by y4my4my4m** [comment]
@mudler 
Was actually going to make the same post.

For my use case, it'd be nice if localai actually was able to make requests to openai using the API key.
Meaning, if you set the model to one of openai's model, it would actually go and make the request to their server.
